// components/MyComponent/MyComponent.js
import {Component, createElement as h} from 'react';
import PropTypes from 'prop-types';


export default class MyComponent extends Component {
	static propTypes = {
		myProp: PropTypes.string.isRequired,
		myOtherProp: PropTypes.string.isRequired,
		onButtonClick: PropTypes.func
	};
	static defaultProps = {
		myProp: 'Default myprop value',
		myOtherProp: 'Default other prop'
	};
	state = {
		myState: 1
	};
	mybutton = null;

	clickHandler = () => {
		if (this.props.onButtonClick){
			this.props.onButtonClick();
		}
		this.setState({
			myState: this.state.myState + 1
		});
	};

	render(){
		const {myProp, myOtherProp} = this.props;
		const {myState} = this.state;
		const saveButton = el => {
			this.mybutton = el;
		};
		return h(
			'div', {className: 'MyComponent'},
			h('p', null, 'myProp: ', myProp),
			h('p', null, 'myOtherProp: ', myOtherProp),
			h('p', null, 'myState: ', myState),
			h('p', null, 'Hello World'),
			h('button', {ref: saveButton, onClick: this.clickHandler}, 'Button with a ref')
		);
	}
}
